src/daft-core/src/array/ops/cast.rs

```diff
@@ -186,21 +186,13 @@ where
         )?
     } else if can_cast_types(&self_arrow_type, &target_arrow_type) {
         // Cast from logical Arrow2 type to logical Arrow2 type.
-        let arrow_logical = cast(
+        cast(
             to_cast.data(),
             &target_arrow_type,
             CastOptions {
                 wrapped: true,
                 partial: false,
             },
-        )?;
-        cast(
```

Comment:
@clarkzinzow do you know why this is necessary? Removing it seems to still keep all our other tests green, and I'm not sure why we would be casting from `logical -> physical` type in this circumstance?